packages/core/src/api.ts

```diff
@@ -239,16 +239,20 @@ const updateStateWithFetchResults = async (
   workspace: Workspace,
   mergedElements: Element[],
   unmergedElements: Element[],
-  fetchedAccounts: string[]
+  fetchedAccounts: string[],
+  partiallyFetchedAccounts: string[],
 ): Promise<void> => {
   const fetchElementsFilter = shouldElementBeIncluded(fetchedAccounts)
   const stateElementsNotCoveredByFetch = await awu(await workspace.state().getAll())
     .filter(element => !fetchElementsFilter(element.elemID)).toArray()
   await workspace.state()
     .override(awu(mergedElements)
       .concat(stateElementsNotCoveredByFetch), fetchedAccounts)
-  await workspace.state().updatePathIndex(unmergedElements,
-    (await workspace.state().existingAccounts()).filter(key => !fetchedAccounts.includes(key)))
+  const accountsToMaintain = [
+    ...(await workspace.state().existingAccounts()).filter(key => !fetchedAccounts.includes(key)),
+    ...partiallyFetchedAccounts,
+  ]
```

Comment:
should not be very large lists, but we should still try to avoid using `...` to concat lists when possible
```suggestion
  const accountsToMaintain = partiallyFetchedAccounts
    .concat(await workspace.state().existingAccounts()).filter(key => !fetchedAccounts.includes(key))
```